Src/Plugins/Security/SEC101_054.AlibabaAccessKeyValidator.cs

```diff
@@ -0,0 +1,124 @@
+        {
+            return IsValidStatic(Instance,
+                                 ref matchedPattern,
+                                 ref groups,
+                                 ref failureLevel,
+                                 ref fingerprint,
+                                 ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message)
+        {
+            return IsValidDynamic(Instance,
+                                   ref fingerprint,
+                                   ref message);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            if (!groups.TryGetNonEmptyValue("account", out string account) ||
+                !groups.TryGetNonEmptyValue("password", out string password))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Platform = nameof(AssetPlatform.AlibabaCloud),
+                Account = account,
+                Password = password,
+            }.ToString();
+
+            return nameof(ValidationState.Unknown);
+        }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText,
+                                                       ref string message)
+        {
+            var fingerprint = new Fingerprint(fingerprintText);
+            string account = fingerprint.Account;
+            string password = fingerprint.Password;
+
+            try
+            {
+                // Taken from https://www.alibabacloud.com/help/doc-detail/63638.htm
+                IClientProfile clientProfile = DefaultProfile.GetProfile("cn-shanghai", account, password);
+                var client = new DefaultAcsClient(clientProfile);
+
+                var request = new PubRequest();
+
+                // We don't actually care about any of the product stuff, so just leave dummy values in.
+                request.ProductKey = "<productKey>";
+                request.TopicFullName = "/<productKey>/<deviceName>/get";
+                byte[] payload = Encoding.Default.GetBytes("Invalid payload.");
+                string payloadStr = Convert.ToBase64String(payload);
+                request.MessageContent = payloadStr;
+                request.Qos = 0;
+                PubResponse response = client.GetAcsResponse(request);
+
+                if (!response.ErrorMessage.Equals(ProductKeyInvalidFormat))
+                {
+                    message = "Unexpected response from Alibaba Cloud.";
```

Comment:
>Unexpected response from Alibaba Cloud [](start = 31, length = 38)

Always put relevant data in the message when reporting something unexpected. This helps with troubleshooting directly from scan reporting. Note my use of apostrophes below to encapsulate the dynamic data not 'owned' by us, i.e., provided by the 3rd party system.

message = $"Unexpected response from Alibaba Cloud: '{response.ErrorMessage}'";